src/main/java/uk/gov/laa/ccms/caab/service/LookupService.java

```diff
@@ -236,6 +244,112 @@ public Mono<CommonLookupDetail> getApplicationStatuses() {
     return ebsApiClient.getCommonValues(COMMON_VALUE_APPLICATION_STATUS);
   }
 
+  /**
+   * Get a list of matter types for proceedings.
+   *
+   * @return MatterTypeLookupDetail containing the matterType values.
+   */
+  public Mono<MatterTypeLookupDetail> getMatterTypes(final String categoryOfLaw) {
+    return ebsApiClient.getMatterTypes(categoryOfLaw);
+  }
+
+  /**
+   * Get a list of Matter Type Common Values.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
+   */
+  public Mono<CommonLookupDetail> getMatterTypes() {
+    return ebsApiClient.getCommonValues(COMMON_VALUE_MATTER_TYPES);
+  }
+
+  /**
+   * Get a list of proceeding types for proceedings.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
+   */
+  public Mono<ProceedingDetails> getProceedings(
+      final ApplicationDetail application,
+      final String matterType,
+      final Boolean isLead) {
+
+    final String categoryOfLaw = application.getCategoryOfLaw().getId();
+    final Boolean isAmendment = application.getAmendment();
+    final Boolean larScopeFlag = application.getLarScopeFlag();
+    final String applicationType = application.getApplicationType().getId();
+
+
+    return ebsApiClient.getProceedings(
+        categoryOfLaw,
+        matterType,
+        isAmendment,
+        larScopeFlag,
+        applicationType,
+        isLead);
+  }
+
+  /**
+   * Retrieves Client involvement types with detailed parameters.
+   *
+   * @param proceedingCode The proceeding code.
+   * @return A Mono containing the ClientInvolvementTypeLookupDetail or an error handler if an
+   *         error occurs.
+   */
+  public Mono<ClientInvolvementTypeLookupDetail> getClientInvolvementTypes(
+      final String proceedingCode) {
+    return ebsApiClient.getClientInvolvementTypes(proceedingCode);
+  }
+
+  /**
+   * Get a list of Client Involvement Types Common Values.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
+   */
+  public Mono<CommonLookupDetail> getClientInvolvementTypes() {
+    return ebsApiClient.getCommonValues(COMMON_VALUE_CLIENT_INVOLVEMENT_TYPES);
+  }
+
+  /**
+   * Retrieves the level of service types for a given proceeding.
+   * The level of service types are fetched based on the proceeding code, category of law, and
+   * matter type.
+   *
+   * @param application The application detail object which contains the category of law.
+   * @param proceedingCode The code of the proceeding.
+   * @param matterType The type of the matter.
+   * @return A Mono of LevelOfServiceLookupDetail containing the level of service types.
+   */
+  public Mono<LevelOfServiceLookupDetail> getProceedingLevelOfServiceTypes(
+      final ApplicationDetail application,
```

Comment:
Same comment here - could we just pass in the categoryOfLawId?

Reply:
amended